### ISSUE TITLE
'System.Console' threw an exception(embedding mono)

### ISSUE THREAD

**Issue by qinpersevere** [mono/mono#17593]
## Steps to Reproduce
I'm trying to call infer.net dll in c++ on linux, current method is embedding mono, but got errors.
1. compile a C# class library (in Visual Studio 2019), using the FirstExample.cs:
```csharp
using System;
using Microsoft.ML.Probabilistic.Models;
namespace Test
{
    public class Test
    {
        public void test()
        {
            Variable<bool> firstCoin = Variable.Bernoulli(0.5).Named("firstCoin");
            Variable<bool> secondCoin = Variable.Bernoulli(0.5).Named("secondCoin");
            Variable<bool> bothHeads = (firstCoin & secondCoin).Named("bothHeads");
            InferenceEngine engine = new InferenceEngine();
            Console.WriteLine("Probability both coins are heads: " + engine.Infer(bothHeads));
            bothHeads.ObservedValue = false;
            Console.WriteLine("Probability distribution over firstCoin: " + engine.Infer(firstCoin));
        }
    }
}
```
2. use the following command to compile embedding mono code:
g++ test.cpp `pkg-config --cflags --libs mono-2`
3. MONO_LOG_LEVEL=debug ./a.out and got errors:

### Current Behavior
As the title says, instantly System.Console exception.

## On which platforms did you notice this

[ ] macOS
[**] Linux
[ ] Windows

**Version Used**:
mono: 6.4.0.198
dotnet:2.1.802

## Stacktrace
```
Mono: DllImport error loading library 'libSystem.Native': 'libSystem.Native: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library 'libSystem.Native.so': 'libSystem.Native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library 'libSystem.Native.so': 'libSystem.Native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library 'libSystem.Native': 'libSystem.Native: cannot open shared object file: No such file or directory'.
Mono: DllImport unable to load library 'libSystem.Native: cannot open shared object file: No such file or directory'.
Mono: DllImport attempting to load: 'System.Native'.
Mono: DllImport error loading library '/usr/lib/mono/4.5/System.Native': '/usr/lib/mono/4.5/System.Native: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/usr/lib/mono/4.5/libSystem.Native.so': '/usr/lib/mono/4.5/libSystem.Native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/usr/lib/mono/4.5/libSystem.Native.so': '/usr/lib/mono/4.5/libSystem.Native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/data/qinxiaolei/game_matchmaking/cpp/FrameworkTest/bin/lib/System.Native': '/data/qinxiaolei/game_matchmaking/cpp/FrameworkTest/bin/lib/System.Native: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/data/qinxiaolei/game_matchmaking/cpp/FrameworkTest/bin/lib/libSystem.Native.so': '/data/qinxiaolei/game_matchmaking/cpp/FrameworkTest/bin/lib/libSystem.Native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/data/qinxiaolei/game_matchmaking/cpp/FrameworkTest/bin/lib/libSystem.Native.so': '/data/qinxiaolei/game_matchmaking/cpp/FrameworkTest/bin/lib/libSystem.Native.so: cannot open shared object file: No such file or directory'.

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.DllNotFoundException: System.Native assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) Interop+Sys.Stat(byte&,Interop/Sys/FileStatus&)
  at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.Directory.Exists (System.String path) [0x0001e] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.TermInfoDriver.SearchTerminfo (System.String term) [0x00020] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.TermInfoDriver..ctor (System.String term) [0x0004b] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.ConsoleDriver..cctor () [0x0004d] in <285579f54af44a2ca048dad6be20e190>:0 
   --- End of inner exception stack trace ---
  at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x00007] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.Console..cctor () [0x0007d] in <285579f54af44a2ca048dad6be20e190>:0 
   --- End of inner exception stack trace ---
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.DllNotFoundException: System.Native assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) Interop+Sys.Stat(byte&,Interop/Sys/FileStatus&)
  at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.IO.Directory.Exists (System.String path) [0x0001e] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.TermInfoDriver.SearchTerminfo (System.String term) [0x00020] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.TermInfoDriver..ctor (System.String term) [0x0004b] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.ConsoleDriver..cctor () [0x0004d] in <285579f54af44a2ca048dad6be20e190>:0 
   --- End of inner exception stack trace ---
  at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x00007] in <285579f54af44a2ca048dad6be20e190>:0 
  at System.Console..cctor () [0x0007d] in <285579f54af44a2ca048dad6be20e190>:0 
   --- End of inner exception stack trace ---
```
```
Please paste the stack trace here if available.
```

**Comment by qinpersevere** [comment]
tried to set $TERM=xterm,still got the same error
I'm beginner to Mono, thanks.

**Comment by lewurm** [comment]
Hi @qinpersevere! 

1. How does `test.cpp` look like?
1. How did you install mono? What distribution do you use?
1. What's the content of `/etc/mono/config`?
1. Does `/usr/lib/libmono-native.so` exist?

**Comment by qinpersevere** [comment]
@lewurm thanks for reply.

1.test.cpp is the embedding mono code, which looks like:
![cpp](https://user-images.githubusercontent.com/22725102/67766486-80360c00-fa89-11e9-8adc-392c99ee29fb.png)
2.I installed mono with reference to the official tutorial, all the tests are based on ubuntu16.04, mono 6.4.0.198. I also tested these on docker, and got the same result.
3. the content of /etc/mono/config:
![content](https://user-images.githubusercontent.com/22725102/67767127-eff8c680-fa8a-11e9-98d2-c9c1f58a21ef.png)
4. yes

is this error related to infer.net?

**Comment by lewurm** [comment]
@qinpersevere I think your embedding code is missing an API call during initialisation:
```c
  mono_config_parse ("/etc/mono/config");
```
could you add it somewhere before `mono_jit_init ()`?

**Comment by qinpersevere** [comment]
@lewurm there is no difference after doing this:
`#include <mono/metadata/mono-config.h>`
```
const char *dlllib = "./FrameworkTest.dll";
mono_config_parse("/etc/mono/config");
MonoDomain *domain = mono_jit_init("Mydomain");
...
```
still trying...

**Comment by qinpersevere** [comment]
hi @lewurm, to be specific, there is a slight difference in the error message:
```
Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.DllNotFoundException: /libmono-native.so assembly:<unknown assembly> type:<unknown type> member:(null)
```
```
Mono: DllImport attempting to load: '/libmono-native.so'.
Mono: DllImport error loading library '/libmono-native.so': '/libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '//libmono-native.so': '//libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '//libmono-native.so': '//libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '//libmono-native.so': '//libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/libmono-native.so': '/libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport unable to load library '/libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport attempting to load: '/libmono-native.so'.
Mono: DllImport error loading library '/libmono-native.so': '/libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '//libmono-native.so': '//libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '//libmono-native.so': '//libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '//libmono-native.so': '//libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/libmono-native.so': '/libmono-native.so: cannot open shared object file: No such file or directory'.
Mono: DllImport unable to load library '/libmono-native.so: cannot open shared object file: No such file or directory'.
```
which looks smilar to #16991 , but it's a different task.
meanwhile, there is no error to run a demo c# code without infer.net in the same way.
thanks.

**Comment by qinpersevere** [comment]
fogot to say, this error can be solved by the same way in #16991 
much appreciate.
